src/ansys/fluent/core/services/app_utilities.py

```diff
@@ -0,0 +1,247 @@
+        return self._stub.IsSolutionDataAvailable(request, metadata=self._metadata)
+
+    def register_pause_on_solution_events(
+        self, request: AppUtilitiesProtoModule.RegisterPauseOnSolutionEventsRequest
+    ) -> AppUtilitiesProtoModule.RegisterPauseOnSolutionEventsResponse:
+        """Register on pause solution events RPC of AppUtilities service."""
+        return self._stub.RegisterPauseOnSolutionEvents(
+            request, metadata=self._metadata
+        )
+
+    def resume_on_solution_event(
+        self, request: AppUtilitiesProtoModule.ResumeOnSolutionEventRequest
+    ) -> AppUtilitiesProtoModule.ResumeOnSolutionEventResponse:
+        """Resume on solution event RPC of AppUtilities service."""
+        return self._stub.ResumeOnSolutionEvent(request, metadata=self._metadata)
+
+    def unregister_pause_on_solution_events(
+        self, request: AppUtilitiesProtoModule.UnregisterPauseOnSolutionEventsRequest
+    ) -> AppUtilitiesProtoModule.UnregisterPauseOnSolutionEventsResponse:
+        """Unregister on pause solution events RPC of AppUtilities service."""
+        return self._stub.UnregisterPauseOnSolutionEvents(
+            request, metadata=self._metadata
+        )
+
+    def exit(
+        self, request: AppUtilitiesProtoModule.ExitRequest
+    ) -> AppUtilitiesProtoModule.ExitResponse:
+        """Exit RPC of AppUtilities service."""
+        return self._stub.Exit(request, metadata=self._metadata)
+
+
+class AppUtilities:
+    """AppUtilities."""
+
+    def __init__(self, service: AppUtilitiesService):
+        """__init__ method of AppUtilities class."""
+        self.service = service
+
+    def get_product_version(self) -> Any:
+        """Get product version."""
+        request = AppUtilitiesProtoModule.GetProductVersionRequest()
+        response = self.service.get_product_version(request)
+        return f"{response.major}.{response.minor}.{response.patch}"
+
+    def get_build_info(self) -> Any:
+        """Get build info."""
+        request = AppUtilitiesProtoModule.GetBuildInfoRequest()
+        response = self.service.get_build_info(request)
+        return f"Build Time: {response.build_time} Build ID: {response.build_id} Revision: {response.vcs_revision} Branch: {response.vcs_branch}"
+
+    def get_controller_process_info(self) -> Any:
+        """Get controller process info."""
+        request = AppUtilitiesProtoModule.GetControllerProcessInfoRequest()
+        response = self.service.get_controller_process_info(request)
+        return {
+            "hostname": response.hostname,
+            "process_id": response.process_id,
+            "working_directory": response.working_directory,
+        }
+
+    def get_solver_process_info(self) -> Any:
+        """Get solver process info."""
+        request = AppUtilitiesProtoModule.GetSolverProcessInfoRequest()
+        response = self.service.get_solver_process_info(request)
```

Comment:
Following code will avoid manually building the dictionary (see usage in exisiting code)
```
from google.protobuf.json_format import MessageToDict
return MessageToDict(response)
```

Reply:
Done. Thank you.

Reply:
@mkundu1 MessageToDict converts all values to strings. It converted integer `process_id` to string as follows and therefore Fluent connection was terminating without proper error message. It has taken some time to investigate the exact cause of error but we have identified it now.

![image](https://github.com/user-attachments/assets/76c04bdc-d493-49fb-b90e-53b66d699451)

![image](https://github.com/user-attachments/assets/80512992-969f-4d69-8527-b4b4b4d8d3df)

We are creating dictionary manually now.